File: fastify/src/passwords.ts
import argon2, { argon2id } from "argon2";
import os from "os";

export default class Passwords {
  #CONFIG: argon2.Options = {
    type: argon2id,
    memoryCost: 128 * 1024, // 128mb
    parallelism: os.cpus().length,
  };

  async hashPassword(password: string) {
    return await argon2.hash(password, this.#CONFIG);
  }

  async verifyPassword(hash: string, password: string) {
    return await argon2.verify(hash, password);
  }
}

File: Api/Passwords.cs
using BCrypt.Net;


namespace Passwords;

static class Password
{
    public static string Hash(string password)
    {
        string salt = BCrypt.Net.BCrypt.GenerateSalt();
        string hash = BCrypt.Net.BCrypt.HashPassword(password, salt);

        return hash;
    }

    public static bool Verify(string password, string hash)
    {
        return BCrypt.Net.BCrypt.Verify(password, hash);
    }
}